h/views/oidc.py

```diff
@@ -100,21 +140,41 @@ def redirect(self):
             # We received an invalid or unexpected redirect from ORCID.
             raise
 
-        orcid = self._orcid_client.get_orcid(validated_params["code"])
+        decoded_state = _decode_oauth2_state_param(
+            validated_params["state"], state_signing_key
+        )
```

Comment:
I'm not handling the exceptions that PyJWT might raise if the JWT fails to decode. By this point we've already verified that the encoded JWT from ORCID matches the copy that we stashed in the browser session so I think it's very unlikely that it would fail to decode.